machinelearning/onnx.md

```diff
@@ -0,0 +1,102 @@
+# ML inference on Graviton CPUs with Open Neural Network Exchange(ONNX)
+
+**Introduction**
+
+ONNX is an open-source machine learning framework that provides interoperability between different frameworks. ONNX Runtime is the runtime engine used for model inference and training with ONNX . This document covers how to use ONNX based machine learning inference on Graviton CPUs, what runtime configurations are important and how to run benchmarking scripts. The document also covers instructions for source builds to enable experimenting with downstream features.
+
+# How to use ONNX on Graviton CPUs
+
+Python wheels is the recommended option to use ONNX on Graviton, the default backend is optimized for Graviton CPU.
+
+```
+# Install Python
+sudo apt-get update
+sudo apt-get install -y python3 python3-pip
+
+# Upgrade pip3 to the latest version
+python3 -m pip install --upgrade pip
+
+# Install ONNX and ONNX Runtime
+python3 -m pip install onnx
+python3 -m pip install onnxruntime
+```
+
+# Prerequisites
+
+1. It is highly recommended to use the AMIs based on Linux Kernel 5.10 and beyond for the best Onnx inference performance on Graviton3 instances
```

Comment:
Would it be helpful to list a few of the most common AMIs which meet this requirement?

Reply:
since new AMIs keep coming and this user guide is valid for different services, I'm thinking listing some AMI might not be good. Instead, I'm planning to add a query to get the latest AMIs that meet the kernel requirement.

Reply:
`--image-id resolve:ssm:/aws/service/ami-amazon-linux-latest/amzn2-ami-kernel-5.10-hvm-arm64-gp2`